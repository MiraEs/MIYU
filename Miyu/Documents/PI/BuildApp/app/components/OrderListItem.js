import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	TouchableOpacity,
	View,
} from 'react-native';
import styles from '../lib/styles';
import {
	Image,
	Text,
} from 'BuildLibrary';
import Icon from 'react-native-vector-icons/Ionicons';
import helpers from '../lib/helpers';
import { IMAGE_75 } from '../constants/CloudinaryConstants';

const componentStyles = StyleSheet.create({
	container: {
		marginTop: styles.measurements.gridSpace1,
		backgroundColor: styles.colors.white,
		padding: styles.measurements.gridSpace1,
	},
	titleBar: {
		paddingBottom: styles.measurements.gridSpace1,
		borderBottomColor: styles.colors.greyLight,
		borderBottomWidth: styles.dimensions.borderWidth,
	},
	body: {
		paddingTop: styles.measurements.gridSpace1,
		flexDirection: 'row',
	},
	contentBody: {
		paddingLeft: styles.measurements.gridSpace1,
		flex: 1,
		justifyContent: 'space-between',
		alignItems: 'center',
		flexDirection: 'row',
	},
	projectIcon: {
		marginRight: styles.measurements.gridSpace1,
	},
});

export default class OrderListItem extends Component {

	defaultTitleBar = () => {
		return (
			<Text weight="bold">{this.props.title}</Text>
		);
	}

	renderTitleLine(titleLine) {
		const titleLineContent = (
			<Text
				lineHeight={false}
				weight="bold"
			>
					{titleLine}
			</Text>
		);

		if (this.props.projectName) {
			return (
				<View style={styles.elements.flexRow}>
					<Icon
						name="md-folder"
						size={16}
						color={styles.colors.secondary}
						style={componentStyles.projectIcon}
					/>
					{titleLineContent}
				</View>
			);
		}

		return titleLineContent;
	}

	render() {
		const { details, image, projectName, renderHeader } = this.props;
		const adjustedDetails = [...details];
		let titleLine = projectName;
		if (!projectName) {
			titleLine = adjustedDetails.shift();
		}

		return (
			<View style={componentStyles.container}>
				<View style={componentStyles.titleBar}>
					{renderHeader ? renderHeader() : this.defaultTitleBar()}
				</View>
				<TouchableOpacity
					style={componentStyles.body}
					onPress={this.props.onPress}
				>
					<Image
						source={image}
						{...IMAGE_75}
					/>
					<View style={componentStyles.contentBody}>
						<View>
							{this.renderTitleLine(titleLine)}
							{adjustedDetails.map((detail, i) =>
								<Text key={i}>{detail}</Text>
							)}
						</View>
						<Icon
							name={helpers.getIcon('arrow-forward')}
							size={25}
							color={styles.colors.grey}
						/>
					</View>
				</TouchableOpacity>
			</View>
		);
	}
}

OrderListItem.propTypes = {
	details: PropTypes.array.isRequired,
	image: PropTypes.object.isRequired,
	onPress: PropTypes.func.isRequired,
	projectName: PropTypes.string,
	renderHeader: PropTypes.func,
	title: PropTypes.string,
};

OrderListItem.defaultProps = {
	title: 'Title',
};
