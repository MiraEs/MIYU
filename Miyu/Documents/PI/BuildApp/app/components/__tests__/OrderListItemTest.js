jest.mock('react-native');
jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

import React from 'react';
import OrderListItem from '../OrderListItem';

const defaultProps = {
	details: ['1', '2', '3'],
	image: {},
	onPress: jest.fn(),
	projectName: 'test',
	renderHeader: jest.fn(),
	title: 'title',
};

describe('OrderListItem component', () => {
	it('should render OrderListItem with projectName', () => {
		const tree = require('react-test-renderer').create(
			<OrderListItem {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render OrderListItem without projectName', () => {
		const tree = require('react-test-renderer').create(
			<OrderListItem
				{...defaultProps}
				projectName={undefined}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render default header', () => {
		const tree = require('react-test-renderer').create(
			<OrderListItem
				{...defaultProps}
				renderHeader={undefined}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
