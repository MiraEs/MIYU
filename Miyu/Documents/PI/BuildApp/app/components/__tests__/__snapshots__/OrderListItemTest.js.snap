// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderListItem component should render OrderListItem with projectName 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "marginTop": 8,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EBEDEF",
        "borderBottomWidth": 1,
        "paddingBottom": 8,
      }
    }
  />
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 8,
      }
    }
  >
    <Image
      height={75}
      source={Object {}}
      width={75}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Icon
            color="#63666A"
            name="md-folder"
            size={16}
            style={
              Object {
                "marginRight": 8,
              }
            }
          />
          <Text
            lineHeight={false}
            weight="bold"
          >
            test
          </Text>
        </View>
        <Text>
          1
        </Text>
        <Text>
          2
        </Text>
        <Text>
          3
        </Text>
      </View>
      <Icon
        color="#D0D3D4"
        name="ios-arrow-forward"
        size={25}
      />
    </View>
  </TouchableOpacity>
</View>
`;

exports[`OrderListItem component should render OrderListItem without projectName 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "marginTop": 8,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EBEDEF",
        "borderBottomWidth": 1,
        "paddingBottom": 8,
      }
    }
  />
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 8,
      }
    }
  >
    <Image
      height={75}
      source={Object {}}
      width={75}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
        }
      }
    >
      <View>
        <Text
          lineHeight={false}
          weight="bold"
        >
          1
        </Text>
        <Text>
          2
        </Text>
        <Text>
          3
        </Text>
      </View>
      <Icon
        color="#D0D3D4"
        name="ios-arrow-forward"
        size={25}
      />
    </View>
  </TouchableOpacity>
</View>
`;

exports[`OrderListItem component should render default header 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "marginTop": 8,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EBEDEF",
        "borderBottomWidth": 1,
        "paddingBottom": 8,
      }
    }
  >
    <Text
      weight="bold"
    >
      title
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 8,
      }
    }
  >
    <Image
      height={75}
      source={Object {}}
      width={75}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 8,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Icon
            color="#63666A"
            name="md-folder"
            size={16}
            style={
              Object {
                "marginRight": 8,
              }
            }
          />
          <Text
            lineHeight={false}
            weight="bold"
          >
            test
          </Text>
        </View>
        <Text>
          1
        </Text>
        <Text>
          2
        </Text>
        <Text>
          3
        </Text>
      </View>
      <Icon
        color="#D0D3D4"
        name="ios-arrow-forward"
        size={25}
      />
    </View>
  </TouchableOpacity>
</View>
`;
